### ISSUE TITLE
No way to turn off variable interpolation or other advanced features

### ISSUE THREAD

**Issue by haltman-at** [ljharb/shell-quote#7]
You can *kind* of turn off variable interpolation by passing `x => "$" + x` for `env`, but not really, because then `"${THING}"` will get turned into `"$THING"` instead of remaining as `"${THING}"`.

Also, it would be nice to have a way to turn off things like operator, comment, and glob handling as well, and just treat these as ordinary strings.  (Glob handling isn't even documented at all, and so is a potential landmine for users who don't read the code...)

Thanks!

**Comment by ljharb** [comment]
Are there any shells where these things are off?

**Comment by haltman-at** [comment]
No, but shell-quote handling is useful outside of the context of shells proper, in contexts where these things don't make sense.

**Comment by ljharb** [comment]
What sort of contexts? I'm not sure it makes sense to add complexity to a package called "shell quote" for non-shell usage.

**Comment by haltman-at** [comment]
I mean basically in our context what's going on is we have something that provides a shell-like interface for commands, but doesn't have anywhere near the power of a real shell, such that attempting to handle operators, etc, doesn't make sense, as we won't be able to do anything with them.

**Comment by ljharb** [comment]
Does that mean you don't even want to parse them, as opposed to failing loudly when they try to use them?

**Comment by ljharb** [comment]
Closing due to lack of followup; happy to reopen if that changes.

**Comment by haltman-at** [comment]
Wow, I never responded to this, huh?  Yes, my intent was that anything other than quote characters and escape characters was treated normally and had no special meaning.  So no it wouldn't error, it'd just be treated like a normal character.

But, uh, also, the project that I was employed as the maintainer of that I was trying to use this in is now defunct.  So it's certainly not relevant to *me*, or any of us, anymore...

**Comment by ljharb** [comment]
gotcha. if nobody else has the use case, then let's not add it until they do :-)
